broadcast/transaction.go

```diff
@@ -30,14 +36,37 @@ func WithCallback(callbackURL string, callbackToken ...string) TransactionOptFun
 	}
 }
 
+// WithMerkleProof will return merkle proof from Arc.
 func WithMerkleProof() TransactionOptFunc {
 	return func(o *TransactionOpts) {
 		o.MerkleProof = true
 	}
 }
 
+// WithWaitForStatus will allow you to return the result only
+// when the transaction reaches the status you set.
 func WithWaitForStatus(status TxStatus) TransactionOptFunc {
 	return func(o *TransactionOpts) {
 		o.WaitForStatus = status
 	}
 }
+
+// WithBeefFormat will accept your transaction in BEEF format 
+// and decode it for a proper format acceptable by Arc.
+func WithBeefFormat() TransactionOptFunc {
+	return func(o *TransactionOpts) {
+		o.BeefFormat = true
+		o.RawFormat = false
+	}
+}
+
+// WithRawFormat will accept your transaction in RawTx format 
+// and encode it for a proper format acceptable by Arc.
+// Deprecated: This function will be depreacted soon.
+// Only EF and BEEF format will be acceptable.
+func WithRawFormat() TransactionOptFunc {
```

Comment:
We need a similar func for EF, right?

Reply:
I thought we don't because it's default, but you're right, without it we can have unnecessary confusions, so I added it.